third_party/nvfuser/csrc/ops/arith.cpp

```diff
@@ -1347,6 +1347,41 @@ TensorView* sum(
   return reductionOp(BinaryOpType::Add, axes, init, v1, keep_dim, dtype);
 }
 
+TensorView* prod(
+    TensorView* v1,
+    const std::vector<int>& axes,
+    bool keep_dim /*=false*/,
+    DataType dtype /* DataType::Null */) {
+  if (dtype == DataType::Null) {
+    auto initial_v1_dtype = v1->getDataType().value();
+    if (isBooleanType(initial_v1_dtype) || isIntegralType(initial_v1_dtype)) {
+      dtype = DataType::Int;
+    }
+  }
+
+  // Cast input tensor to dtype before the operation is performed
+  if (dtype != DataType::Null) {
+    v1 = optionalCastStrict(dtype, v1)->as<TensorView>();
+  }
+
+  Val* init = nullptr;
```

Comment:
Should this `init` block be refactored like `ops::getMinimumValue` or `ops::getMaximumValue`? I see that `sum` has a similar pattern too.

Reply:
Yeah possibly. We do have `Fusion::oneVal()` and `Fusion::zeroVal()` but they are scalar `Double`s. A simple solution would be to add an optional `dtype` argument to `oneVal` and `zeroVal`. I'll try it.